homeassistant/components/sonos/alarms.py

```diff
@@ -23,48 +23,72 @@ class SonosAlarms(SonosHouseholdCoordinator):
     def __init__(self, *args: Any) -> None:
         """Initialize the data."""
         super().__init__(*args)
-        self._alarms: dict[str, Alarm] = {}
+        self.alarms: Alarms = Alarms()
+        self.created_alarm_ids: set[str] = set()
 
     def __iter__(self) -> Iterator:
         """Return an iterator for the known alarms."""
-        alarms = list(self._alarms.values())
-        return iter(alarms)
+        return iter(self.alarms)
 
     def get(self, alarm_id: str) -> Alarm | None:
         """Get an Alarm instance."""
-        return self._alarms.get(alarm_id)
+        return self.alarms.get(alarm_id)
 
-    async def async_update_entities(self, soco: SoCo) -> bool:
+    async def async_update_entities(
+        self, soco: SoCo, update_id: int | None = None
+    ) -> None:
         """Create and update alarms entities, return success."""
-        try:
-            new_alarms = await self.hass.async_add_executor_job(self.update_cache, soco)
-        except (OSError, SoCoException) as err:
-            _LOGGER.error("Could not refresh alarms using %s: %s", soco, err)
-            return False
-
-        for alarm in new_alarms:
-            speaker = self.hass.data[DATA_SONOS].discovered[alarm.zone.uid]
-            async_dispatcher_send(
-                self.hass, SONOS_CREATE_ALARM, speaker, [alarm.alarm_id]
-            )
+        updated = await self.hass.async_add_executor_job(
+            self.update_cache, soco, update_id
```

Comment:
Can this no longer throw `(OSError, SoCoException)`?

Reply:
Yes, it's still possible. Will catch during the update.